sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlSchemaCheck.java

```diff
@@ -223,48 +230,63 @@ public void setSchemas(String schemas) {
   }
 
   private void validate() {
-    if ("autodetect".equalsIgnoreCase(schemas)) {
-      autodetectSchemaAndValidate();
-    } else {
-      String[] schemaList = StringUtils.split(schemas, " \t\n");
-
-      validate(schemaList);
+    try {
+      if ("autodetect".equalsIgnoreCase(schemas)) {
+        autodetectSchemaAndValidate();
+      } else {
+        String[] schemaList = StringUtils.split(schemas, " \t\n");
+        validate(schemaList);
+      }
+    } catch (SchemaNotFoundException e) {
+      LOG.warn("Cannot validate file {}", getWebSourceCode());
+      LOG.warn("Cause: {}", e.toString());
     }
   }
 
   private void validate(String[] schemaList) {
-    // Create a new validator
-    Validator validator = createSchema(schemaList).newValidator();
-    setFeature(validator, Constants.XERCES_FEATURE_PREFIX + "continue-after-fatal-error", true);
-    validator.setErrorHandler(new MessageHandler());
-    validator.setResourceResolver(new SchemaResolver());
+    Schema schema = createSchema(schemaList);
+    if (schema != null) {
+      Validator validator = createSchema(schemaList).newValidator();
+      setFeature(validator, Constants.XERCES_FEATURE_PREFIX + "continue-after-fatal-error", true);
+      validator.setErrorHandler(new MessageHandler());
+      validator.setResourceResolver(new SchemaResolver());
 
-    // Validate and catch the exceptions. MessageHandler will receive the errors and warnings.
-    try {
-      if (LOG.isInfoEnabled()) {
+      // Validate and catch the exceptions. MessageHandler will receive the errors and warnings.
+      try {
         LOG.info("Validating {} with schema {}", getWebSourceCode(), StringUtils.join(schemaList, ","));
-      }
-      validator.validate(new StreamSource(getWebSourceCode().createInputStream()));
+        validator.validate(new StreamSource(getWebSourceCode().createInputStream()));
 
-    } catch (SAXException e) {
-      if (!containsMessage(e)) {
-        createViolation(0, e.getMessage());
-      }
+      } catch (SAXException e) {
+        if (!containsMessage(e)) {
+          createViolation(0, e.getMessage());
+        }
 
-    } catch (IOException e) {
-      throw new IllegalStateException(e);
+      } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)